libnmstate/nm/connection.py

```diff
@@ -196,16 +193,14 @@ def create_new_nm_simple_conn(iface, nm_profile):
     if team_setting:
         settings.append(team_setting)
 
-    if VRF.CONFIG_SUBTREE in iface_info:
-        settings.append(create_vrf_setting(iface_info[VRF.CONFIG_SUBTREE]))
+    if iface.type == InterfaceType.VRF:
+        settings.append(create_vrf_setting(iface))
 
-    if MacVlan.CONFIG_SUBTREE in iface_info:
-        settings.append(create_macvlan_setting(iface_info, nm_profile))
+    if iface.type == InterfaceType.MAC_VLAN:
+        settings.append(create_macvlan_setting(iface, nm_profile))
 
-    if MacVtap.CONFIG_SUBTREE in iface_info:
-        settings.append(
-            create_macvlan_setting(iface_info, nm_profile, tap=True)
-        )
+    if iface.type == InterfaceType.MAC_VTAP:
+        settings.append(create_macvlan_setting(iface, nm_profile, tap=True))
```

Comment:
This is incorrect. The `create_mactap_setting()` is taking dict instead of `BaseIface` object.

And it should be __vtap__ not __vlan__

Reply:
In order to re-use code we are using the same function for macvlan and macvtap. I did not modify the if statement logic. About taking dict instead of `BaseIface` object, this patch is modifying that function.

Reply:
Oh I forgot that. Then we are good.